code/datums/uplink_item.dm

```diff
@@ -612,6 +616,7 @@ var/list/uplink_items = list()
 
 	if(possible_items.len)
 		var/datum/uplink_item/I = pick(possible_items)
+		cost = max(0, I.get_cost(U.job, 0.5))
```

Comment:
doesn't this change the cost of `/datum/uplink_item/badass/random` for future purchases?

Reply:
oh, I see. You're then checking the name of this datum specifically. I wonder if there isn't a better way.

Reply:
I change it here and then change it back in the function spawn_item gets called in, buy(), just before the two places it could possibly return after it's called.

Reply:
This feels like the wrong approach, my gut like Damian's says, don't actually change the cost; just track the cost elsewhere

Reply:
Okay, removed all this and used new approach that can return an uplink item datum itself with the info in it instead.